client/components/tinymce/plugins/simple-payments/dialog/product-image-picker.jsx

```diff
@@ -0,0 +1,129 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+
+import AsyncLoad from 'components/async-load';
+import { getMediaItem } from 'state/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import MediaActions from 'lib/media/actions';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import ProductImage from './product-image';
+import RemoveButton from 'components/remove-button';
+
+class ProductImagePicker extends Component {
+	static propTypes = {
+		featuredImage: PropTypes.object,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+	};
+
+	state = {
+		isSelecting: false,
+	};
+
+	showMediaModal = () => {
+		const { siteId, featuredImage } = this.props;
+
+		if ( featuredImage ) {
+			MediaActions.setLibrarySelectedItems( siteId, [ featuredImage ] );
+		}
+
+		this.setState( { isSelecting: true } );
+	};
+
+	hideMediaModal = () => this.setState( { isSelecting: false } );
+
+	setImage = value => {
+		this.hideMediaModal();
+
+		if ( ! value ) {
```

Comment:
Should we early return here if ID doesn't have a value, or is it okay to pass a falsey value to onChange? eg

`const id = get( input, [ 'value', 'items', 0 ] );`

Reply:
It's ok to pass a falsey value - it will result in not setting any image. I'm actually doing that in `removeCurrentImage` to remove the existing one.